backend/src/main/java/com/google/sps/datastore/UserDatastore.java

```diff
@@ -28,19 +24,49 @@ public void addUser(User user) {
     // Set properties of entity based on user fields
     Entity userEntity = new Entity(KIND_USER, user.getUsername());
     userEntity.setProperty(PROPERTY_USERNAME, user.getUsername());
-    userEntity.setProperty(PROPERTY_TIMESTAMP, user.getTimestamp());
 
     // Insert entity into datastore
     datastore.put(userEntity);
   }
 
-  /** Return String representation of participants for logging purposes */
+  /** Return User Entity from username, or null if entity is not found */
+  @Nullable
+  private Entity getEntityFromUsername(String username) {
+    Key userKey = KeyFactory.createKey(KIND_USER, username);
+    try {
+      return datastore.get(userKey);
+    } catch (EntityNotFoundException e) {
+      return null;
+    }
+  }
+
+  /** Return User from username, or null if user with username not in datastore */
+  @Nullable
+  public User getUserFromUsername(String username) {
+    return getUserFromEntity(getEntityFromUsername(username));
+  }
+
+  /** Return user object from datastore user entity, or null if entity is null */
+  @Nullable
+  private static User getUserFromEntity(Entity entity) {
```

Comment:
I would make entity @NonNull here to avoid the null check and make the caller check.